### ISSUE TITLE
Unify generator and template concepts and ui

### ISSUE THREAD

**Issue by unbalancedparentheses** [unbalancedparentheses/gut#80]
Gut will only have generators. Generators are stored as github repositories. By defaullt **gut new generator_name desired_name** will create a new directory called `desired_name` with all the compiled files inside of it. However inside of the `gut.conf` configuration file the generator will be able to specify if the generated file or files should be created at the current working dir. This option will let us have the same functionality as the one that would have been provided by templates.

The only gotcha that I need to think while I implement this is how we are going to show on `gut find` that a generator creates a file or many files at the current working dir and it does not setup a project. We could use some keyword at the description and if the keyword is present we would print that line of `gut find` output in another color. Another idea would be that generators that create a directory with files inside of it (aka projects) should state this on the description saying the word project. For example cowboy generator will have a description like "cowboy project generator" and cowboy-handler that only creates a cowboy handler file would say "cowboy handler generator" or "cowboy handler file generator".

I don't want to have the template concept on my code or in the command line. I have been thinking about it and it makes gutenberg overcomplicated without any need. It makes me have two UI (`gut new` and `gut template` or however you want to call them) for no reason. Yeah, I know that rails has `rails new` and `rails generate` but after thinking quite a bit I think that at least for a tool like `gut` this two commands can be merged. I think that `gut` should be smart enough to know if the user wants to create a new project or if you want to create only one new file based on the type of generator he is using.

To sum up I do not want duplicated code, different commands/UI/branches/flow. Having a file or even many files created on the current working dir instead of inside a new directory is not a real difference.
